ledger-core/v2/virtual/integration/vdelegatedrequestfinished_test.go

```diff
@@ -0,0 +1,144 @@
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/gen"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/instrumentation/inslogger"
+	"github.com/insolar/assured-ledger/ledger-core/v2/reference"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/integration/utils"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"reflect"
+	"testing"
+	"time"
+)
+
+// 1. Send VStateReport with 2 pending ordered
+// 2. Send execute
+// 3. Send VDelegateRequestFinished
+// 4. Check that execute process request after VDelegateRequestFinished
+func TestVirtual_SendVStateReport_And_VDelegateRequestFinished(t *testing.T) {
+	server := utils.NewServer(t)
+	ctx := inslogger.TestContext(t)
+
+	testBalance := uint32(100)
+	rawWalletState := makeRawWalletState(t, testBalance)
+	objectRef := gen.Reference()
+
+	proto := testwalletProxy.GetPrototype()
+	pulseNumber := server.GetPulse().PulseNumber
+	stateRef := gen.UniqueIDWithPulse(pulseNumber)
+
+	sr := payload.VStateReport{
+		Polymorph:           uint32(payload.TypeVStateReport),
+		MutablePendingCount: 1,
+		Callee:              objectRef,
+		ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+			LatestDirtyState: &payload.ObjectState{
+				Reference: stateRef,
+				Prototype: proto,
+				State:     rawWalletState,
+			},
+		},
+	}
+
+	var callFlags payload.CallRequestFlags
+
+	callFlags.SetTolerance(payload.CallTolerable)
+
+	cr := payload.VCallRequest{
+		Polymorph:           uint32(payload.TypeVCallRequest),
+		CallType:            payload.CTMethod,
+		CallFlags:           callFlags,
+		CallAsOf:            0,
+		Caller:              server.GlobalCaller(),
+		Callee:              objectRef,
+		CallSiteDeclaration: proto,
+		CallSiteMethod:      "Accept",
+		CallRequestFlags:    0,
+		CallOutgoing:        server.RandomLocalWithPulse(),
+		Arguments:           insolar.MustSerialize([]interface{}{uint32(10)}),
+	}
+
+	df := payload.VDelegatedRequestFinished{
+		Polymorph: uint32(payload.TypeVDelegatedRequestFinished),
+		Callee:    objectRef,
+	}
+
+	requestIsDone := make(chan string, 0)
+
+	server.PublisherMock.Checker = func(topic string, messages ...*message.Message) error {
+		defer func() { requestIsDone <- "callResultReceive" }()
+
+		pl, err := payload.UnmarshalFromMeta(messages[0].Payload)
+		require.NoError(t, err)
+
+		switch pl.(type) {
+		case *payload.VCallResult:
+		default:
+			require.Failf(t, "", "bad payload type, expected %s, got %T", "*payload.VCallResult", pl)
+		}
+		return nil
+	}
+
+	aclSeq := make([]string, 0)
+
+	vStateReportMsg, err := wrapMsg(pulseNumber, &sr)
+	require.NoError(t, err)
+
+	vCallRequestMsg, err := wrapMsg(pulseNumber, &cr)
+	require.NoError(t, err)
+
+	vDelegateRequestFinishedMsg, err := wrapMsg(pulseNumber, &df)
+	require.NoError(t, err)
+
+	server.SendMessage(ctx, vStateReportMsg)
+	aclSeq = append(aclSeq, "stateReportSend")
```

Comment:
I would move all this strings ( "stateReportSend",  "callRequestSend" etc.) to contacts, at least because you use them twice

Reply:
@ivanshibitov Fixed